### PR TITLE
Moved to listing.vue

### PR DESCRIPTION
**Pull request by BobWez98**
Related to #448

### CODE REVIEW

File: resources/js/components/Listing/Listing.vue [hunk 1]
```diff
@@ -96,5 +96,24 @@ export default {
...
+            }
+
+            return {
+                query: {
+                    function_score: {
+                        field_value_factor: {
+                            field: 'positions.'+window.config.category.entity_id,
+                            missing: 0
+                        }
+                    }
```
Comment (Jade-GG) [inline resources/js/components/Listing/Listing.vue:113]: While we're at it, might as well implement the script fix thing? I assume the need for that fix hasn't changed given that this is just a repositioning of the code.

**Review by indykoning — DISMISSED** [review]

File: resources/js/components/Listing/Listing.vue [hunk 2]
```diff
@@ -96,5 +96,25 @@ export default {
             window.history.pushState(window.history.state, '', url)
         },
     },
+
+    methods: {
+        getPositionsQuery() {
```
Comment (royduin) [inline resources/js/components/Listing/Listing.vue:101]:
```suggestion
        getQuery() {
```

File: resources/views/components/listing.blade.php [hunk 3]
```diff
@@ -18,7 +18,7 @@
         }]"
         v-cloak
     >
-        <div slot-scope="{ loaded, filters, sortOptions, reactiveFilters, _renderProxy: listingSlotProps }">
+        <div slot-scope="{ loaded, filters, sortOptions, reactiveFilters, _renderProxy: listingSlotProps, getPositionsQuery }">
```
Comment (royduin) [inline resources/views/components/listing.blade.php:21]:
```suggestion
        <div slot-scope="{ loaded, filters, sortOptions, reactiveFilters, getQuery, _renderProxy: listingSlotProps }">
```

File: resources/views/components/listing.blade.php [hunk 4]
```diff
@@ -29,22 +29,7 @@
...
-                                function_score: {
-                                    field_value_factor: {
-                                        field: 'positions.'+window.config.category.entity_id,
-                                        missing: 0
-                                    }
-                                }
-                            }
-                        }
-                    }"
+                    :custom-query="getPositionsQuery"
```
Comment (royduin) [inline resources/views/components/listing.blade.php:32]:
```suggestion
                    :custom-query="getQuery"
```

File: resources/js/components/Listing/Listing.vue [hunk 5]
```diff
@@ -96,5 +96,25 @@ export default {
...
+            if (!window.config.category?.entity_id) {
+                return;
+            }
+
+            return {
+                query: {
+                    function_score: {
+                        script_score: {
+                            script: {
+                                source: Integer.parseInt(doc['positions.'+(window.config.category.entity_id)].empty ? '0' : doc['positions.'+(window.config.category.entity_id)+''].value),
```
Comment (royduin) [inline resources/js/components/Listing/Listing.vue:111]:
```suggestion
                                source: Integer.parseInt(
                                    doc['positions.' + window.config.category.entity_id].empty
                                        ? '0'
                                        : doc['positions.' + window.config.category.entity_id + ''].value,
                                ),
```

**Pull request merged**